File: Server/src/main/content/global/skill/production/cooking/recipe/cake/ChocolateCake.java
package content.global.skill.production.cooking.recipe.cake;

import content.global.skill.production.cooking.recipe.Recipe;
import core.game.interaction.NodeUsageEvent;
import core.game.node.entity.player.Player;
import core.game.node.entity.skill.Skills;
import core.game.node.item.Item;

/**
 * The Chocolate cake.
 */
public class ChocolateCake extends Recipe {
    private static final Item CAKE = new Item(1891);
    private static final Item CHOCOLATE_CAKE = new Item(1897);
    private static final Item CHOCOLATE_BAR = new Item(1973);

    @Override
    public void mix(final Player player, final NodeUsageEvent event) {
        if (player.getSkills().getLevel(Skills.COOKING) < 50) {
            player.getDialogueInterpreter().sendDialogue("You need a Cooking level of 50 in order to do that.");
            return;
        }
        super.mix(player, event);
        player.getSkills().addExperience(Skills.COOKING, 30, true);
    }

    @Override
    public Item getBase() {
        return CAKE;
    }

    @Override
    public Item getProduct() {
        return CHOCOLATE_CAKE;
    }

    @Override
    public Item[] getIngredients() {
        return new Item[]{CHOCOLATE_BAR};
    }

    @Override
    public Item[] getParts() {
        return new Item[]{};
    }

    @Override
    public String getMixMessage(NodeUsageEvent event) {
        return "You add chocolate to the cake.";
    }

    @Override
    public boolean isSingular() {
        return true;
    }

}

File: Server/src/main/content/global/skill/production/cooking/recipe/cake/ChocolateCake.kt
package content.global.skill.production.cooking.recipe.cake

import content.global.skill.production.cooking.recipe.Recipe
import core.api.consts.Items
import core.api.getStatLevel
import core.api.rewardXP
import core.api.sendDialogue
import core.game.interaction.NodeUsageEvent
import core.game.node.entity.player.Player
import core.game.node.entity.skill.Skills
import core.game.node.item.Item

class ChocolateCake : Recipe() {

    override fun mix(player: Player, event: NodeUsageEvent) {
        if (getStatLevel(player, Skills.COOKING) < 50) {
            sendDialogue(player, "You need a Cooking level of 50 in order to do that.")
            return
        }
        super.mix(player, event)
        rewardXP(player, Skills.COOKING, 30.0)
    }

    override fun getBase(): Item {
        return CAKE
    }

    override fun getProduct(): Item {
        return CHOCOLATE_CAKE
    }

    override fun getIngredients(): Array<Item> {
        return arrayOf(CHOCOLATE_BAR)
    }

    override fun getParts(): Array<Item> {
        return arrayOf()
    }

    override fun getMixMessage(event: NodeUsageEvent): String {
        return "You add chocolate to the cake."
    }

    override fun isSingular(): Boolean {
        return true
    }

    companion object {
        private val CAKE = Item(Items.CAKE_1891)
        private val CHOCOLATE_CAKE = Item(Items.CHOCOLATE_CAKE_1897)
        private val CHOCOLATE_BAR = Item(Items.CHOCOLATE_BAR_1973)
    }
}
